docs/dev/analysis/tbl-cell-borders.rst

```diff
@@ -0,0 +1,92 @@
+
+Table Cell Borders
+------------------
+
+In PowerPoint, a table cell can have borders on all 4 sides, plus two diagonal lines (bottom left to top right and
+top left to bottom right) for a total of 6 possible borders, each specified by a child element of the table cell
+properties element.
+
+Each border is further customized by a set of attributes and child elements common to all borders (`LineFormat`).
+
+In order to provide `LineFormat` functionality, each border `_ln` object is wrapped in a `_TableBorderLine`,
+which implements the `get_or_add_ln` method, and can be then wrapped by `LineFormat`.
+
+Protocol
+--------
+
+**Starting with a default cell** A new cell has no borders or style.
+New borders can be created by accessing the borders property::
+
+    >>> table = shapes.add_table(rows, columns, left, top, width, height).table
+    >>> cell = table.rows[0].cells[0]
+    >>> cell
+    <pptx.shapes.table._Cell object at 0x...>
+    >>> cell.borders
+    <pptx.shapes.table._TableBorders object at 0x...>
```

Comment:
Let's insert here the kinds of interactions you can have with a default `_CellBorders` object and the expected values:

```rst
>>> cell_borders = cell.borders
>>> cell_borders
<pptx.shapes.table._CellBorders object at 0x...>
>>> cell_borders.top
None  # ---or whatever---
>>> ... etc.
```

The return value for an "empty" attribute (on `_CellBorders` in this case) is an important design decision which affects subsequent parts of the protocol and the structure of the "read"-side acceptance tests. For example, if a sub-object like `_CellBorderLine` unconditionally appears on an attribute, whether a line or whatever is actually defined or not, that would appear in an acceptance test like:

```cucumber
Given a _CellBorders object as borders
 Then borders.left is a _CellBorderLine object
  And borders.top is a _CellBorderLine object
  And borders.right is a _CellBorderLine object
  And borders.bottom is a _CellBorderLine object
```

We don't need to spell out all four cases like this in the protocol, one example is enough to demonstrate the behavior, but in the acceptance test they'll all be there because each is a separate piece of the code.

If `_CellBorders` has any other interesting read-side behaviors (like other attributes) we should exercise them here as well.